GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -477,8 +477,11 @@ protected override void OnFormClosed(FormClosedEventArgs e)
                 // a special meaning, and can be dangerous if used inappropriately.
                 if (CommitKind is (CommitKind.Normal or CommitKind.Amend))
                 {
-                    _commitMessageManager.MergeOrCommitMessage = Message.Text;
-                    _commitMessageManager.AmendState = Amend.Checked;
+                    ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+                    {
+                        await _commitMessageManager.SetMergeOrCommitMessageAsync(Message.Text);
+                        await _commitMessageManager.SetAmendStateAsync(Amend.Checked);
+                    });
```

Comment:
```suggestion
                    ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
                    {
                        await _commitMessageManager.SetMergeOrCommitMessageAsync(Message.Text);
                        await _commitMessageManager.SetAmendStateAsync(Amend.Checked);
                    }).FileAndForget();
```

Reply:
That actually has to be sync call, must have been a bad copy paste. Thanks